paddle/fluid/operators/shape_op.cu

```diff
@@ -12,12 +12,58 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License. */
 
+#include <cuda.h>
+#include <vector>
 #include "paddle/fluid/operators/shape_op.h"
 
+namespace paddle {
+namespace operators {
+
+using Tensor = framework::Tensor;
+using LoDTensor = framework::LoDTensor;
+using SelectedRows = framework::SelectedRows;
+
+template <typename T>
+class ShapeCUDAKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext& ctx) const override {
+    auto* in_var = ctx.InputVar("Input");
+    framework::DDim in_dims;
+    if (in_var->IsType<SelectedRows>()) {
+      in_dims = in_var->Get<SelectedRows>().value().dims();
+    } else {
+      in_dims = in_var->Get<LoDTensor>().dims();
+    }
+
+    auto* out_t = ctx.Output<Tensor>("Out");
+    auto dim_size = in_dims.size();
+    out_t->Resize({dim_size});
+    // The data type in DDims is int64_t. Downcasting it into int32_t
+    // to keep same with most usage in framework. This is dangerous
+    // but the data value of shape seems no more than the limit of int32_t.
+    std::vector<int32_t> in_dims_data;
```

Comment:
ddim.h has a function named `vectorize`, could you check whether you can reuse that code?